commands/usernote.js

```diff
@@ -62,7 +70,20 @@ module.exports = {
     };
     let args = parseArgs(preparsedArgs, parseSettings);
     let command = args._.length ? args._[0].toLowerCase() : 'show';
-    args.subreddit = args.link || args.subreddit ? args.subreddit : 'pokemontrades';
+    if (!args.subreddit) {
+      if (moduleConfig.defaultSubreddit) args.subreddit = moduleConfig.defaultSubreddit;
+      if (args.link) {
+        const linkSubMatch = args.link.match(/reddit\.com\/r\/([^\/]+)/);
+        if (linkSubMatch) {
+          args.subreddit = linkSubMatch[1];
+        } else if (args.link.match(/reddit\.com\/message\//)) {
+          args.subreddit = undefined;
+        }
+      }
+    }
+    if (moduleConfig && moduleConfig.channelPermissions && (!moduleConfig.channelPermissions[channel] || (!_.isBoolean(moduleConfig.channelPermissions[channel]) && (args.subreddit===undefined || moduleConfig.channelPermissions[channel].findIndex(sub => sub.toLowerCase() === args.subreddit.toLowerCase()) < 0)))) {
+      throw {error_message: 'Access denied.'};
```

Comment:
Actually, to make the above more concrete, I found a security issue; `.tag undo-delete` only has one stack of removed notes that is shared among all channels. So a user in another channel could use `.tag undo-delete` to re-add notes in subreddits that they don't have access to.